### ISSUE TITLE
Noux/net: renovation of noux/net

### ISSUE THREAD

**Issue by cnuke** [genodelabs/genode#441]
There are a few issues regarding noux/net that make it difficult to deal with. First
it uses its own socketdescriptor registry which needs to be kept in sync with the
normal filedescriptor registry that is provided by the child class. This approach is
quite error-pone because you have to check if the fd you are currently dealing with
is in fact a socketdescriptor when using non-network related syscalls (e.g. close()).
The socketdescriptor registry also needs to be 'flushed' when a child exits which
is rather bad because there is only one(!) sd registry but maybe there are multiple
childs who use it. Furthermore noux/net needs to be notified by the tcp/ip-stack if
there is a event which should be handled (e.g. select()-call on a socket). Therefor
it implements a callback function that is called by the tcp/ip-stack. Within this
function the select() implementation of the tcp/ip-stack is used to determine which
socket is ready. The socketdescriptor registry is than used to lookup the I/O channel
that belongs to the socket so it can be unblocked. This construction is also quite
error-prone.

Long story short, noux/net should use the facilities provided by the child class
(e.g. there should by only one registry of descriptors or rather I/O channels).
In addition the callback-function called from the tcp/ip-stack should only unblock
currently blocked children who are waiting in SYSCALL_SELECT and the tcp/ip-stack's
select() function should only be called when it is really need, (e.g. it should
be called in the Io_channel::check_unblock() method).
